biorxiv/pca_association_experiment/04_biorxiv_word_pca_component_doc2vec.py

```diff
@@ -0,0 +1,194 @@
+
+word_pca_similarity.shape
+
+# +
+word_pca_sim_df = pd.DataFrame(
+    word_pca_similarity,
+    columns=[f"pca{dim}_cossim" for dim in range(1, n_components + 1, 1)],
+).assign(word=model.wv.index2word)
+
+word_pca_sim_df.to_csv(
+    f"output/word_pca_similarity/word_pca_cos_sim_{n_components}_pcs_doc2vec.tsv",
+    sep="\t",
+    index=False,
+)
+
+# Remove those pesky citations from the word pca similarity
+word_pca_sim_df = word_pca_sim_df.query(
+    "~word.str.match(r'^(\(|\[)', na=False)"  # noqa: W605
+)
+word_pca_sim_df.head()
+
+
+# -
+
+# # Generate Word Clouds for the PC dimensions
+
+# Given word to principal component association, next step is to generate word clouds for each principal component. The word clouds have orange representing words that are most similar to the principal component and blue as words most dissimilar to the principal component.
+
+class PolarityColorFunc:
+    def __init__(
+        self,
+        word_class_map,
+        positive_key="positive",
+        positive="#ef8a62",
+        negative_key="negative",
+        negative="#67a9cf",
+        default="#f7f7f7",
+    ):
+        self.positive_class = positive
+        self.negative_class = negative
+        self.positive_key = positive_key
+        self.negative_key = negative_key
+        self.default_class = default
+
+        self.words_to_color_map = word_class_map
+
+    def get_color_mapper(self, word):
+        if word in self.words_to_color_map[self.positive_key]:
+            return ImageColor.getrgb(self.positive_class)
+        # wordcloud.get_single_color_func(self.positive_class)
+        elif word in self.words_to_color_map[self.negative_key]:
+            return ImageColor.getrgb(self.negative_class)
+        else:
+            return ImageColor.getrgb(self.default_class)
+
+    def __call__(self, word, **kwargs):
+        return self.get_color_mapper(word)
+
+
+pca_dimensions = [f"pca{dim}_cossim" for dim in range(1, n_components + 1, 1)]
+for pc, component in tqdm_notebook(enumerate(pca_dimensions, start=1)):
+    word_class_map = {}
+
+    word_class_map["negative"] = (
+        word_pca_sim_df.sort_values(component, ascending=True)
+        .head(100)
+        .assign(**{component: lambda x: x[component].abs().values.tolist()})
+        .assign(**{component: lambda x: x[component] / x[component].max()})[
+            ["word", component]
+        ]
+        .to_dict(orient="records")
+    )
+
+    word_class_map["positive"] = (
+        word_pca_sim_df.sort_values(component, ascending=False)
+        .assign(**{component: lambda x: x[component] / x[component].max()})
+        .head(100)[["word", component]]
+        .to_dict(orient="records")
+    )
+
+    polarity_color_map = PolarityColorFunc(
+        {
+            word_class: set(map(lambda x: x["word"], word_class_map[word_class]))
+            for word_class in word_class_map
+        }
+    )
+
+    pc = f"{pc}" if pc > 9 else f"0{pc}"
```

Comment:
I think you mean this.

```suggestion
    pc = f"{pc:02d}"
```